xbmc/input/InputCodingTableKorean.cpp

```diff
@@ -0,0 +1,371 @@
+            choseong = dicChoseong.find(dicKorean.at(key));
+            jungseong = -1;
+            jongseong = -1;
+          }
+        }
+      }
+      else // If there is no jungseong, new consonant can be choseong or second part of double consonant.
+      {
+        if (choseong == -1) // New consonant is choseong. Also it could be first part of double consonant. 
+        {
+          if (jongseong != -1) // If choseong is already completed, new consonant is another choseong. 
+                               // So previous character has only jongseong.
+          {
+            korean += dicJongseong.at(jongseong);
+            jongseong = -1;
+          }
+          choseong = dicChoseong.find(dicKorean.at(key));
+        }
+        // Find double consonant of chongseong
+        else if (choseong == 0 && key == 9) // ㄳ
+        {			
+          choseong = -1;
+          jongseong = 2;
+        }
+        else if (choseong == 2 && key == 12) // ㄵ
+        {
+          choseong = -1;
+          jongseong = 4;
+        }
+        else if (choseong == 2 && key == 18) // ㄶ
+        {
+          choseong = -1;
+          jongseong = 5;
+        }
+        else if (choseong == 5 && key == 0) // ㄺ
+        {
+          choseong = -1;
+          jongseong = 8;
+        }
+        else if (choseong == 5 && key == 6) // ㄻ
+        {
+          choseong = -1;
+          jongseong = 9;
+        }
+        else if (choseong == 5 && key == 7) // ㄼ
+        {
+          choseong = -1;
+          jongseong = 10;
+        }
+        else if (choseong == 5 && key == 9) // ㄽ
+        {
+          choseong = -1;
+          jongseong = 11;
+        }
+        else if (choseong == 5 && key == 16) // ㄾ
+        {
+          choseong = -1;
+          jongseong = 12;
+        }
+        else if (choseong == 5 && key == 17) // ㄿ
+        {
+          choseong = -1;
+          jongseong = 13;
+        }
+        else if (choseong == 5 && key == 18) // ㅀ
+        {
+          choseong = -1;
+          jongseong = 14;
+        }
+        else if (choseong == 7 && key == 9) // ㅄ
+        {
+          choseong = -1;
+          jongseong = 17;
+        }
+        else // In this case, previous character has only choseong. And new consonant is choseong.
+        {
+          korean += dicChoseong.at(choseong);
+          choseong = dicChoseong.find(dicKorean.at(key));
+        }
+      }
+    }
+    else // If key is vowel, key is jungseong.
+    {
+      if (jongseong != -1) // If previous character has jongseong and this key is jungseong, 
+                           // actually latest vowel is not jongseong. It's choseong of new character.
+      {
+        // If jongseong of previous character is double consonant, we will seperate it to two vowel again. 
+        // First part of double consonant is jongseong of previous character. 
+        // Second part of double consonant is choseong of current character.
+        int newCho;

```

Comment:
Cosmetics, the comments from here downwards are using tabs instead of spaces and not aligned.